gcloud/monitoring/metric.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Metric Descriptors for the Google Monitoring API."""
+
+# Features intentionally omitted from this first version of the client library:
+#   - Creating and deleting metric descriptors.
+
+import collections
+
+from gcloud.monitoring.label import LabelDescriptor
+
+
+class MetricDescriptor(collections.namedtuple(
```

Comment:
Named tuples have been popular among some of my colleagues as a way to make all attributes read-only without having to write out property definitions. Named tuples also give us a useful `__repr__` method, which we need, and meaningful equality comparison, which is sometimes nice.

I'm thinking of sticking with them for simple struct-like things like `Metric`, `Resource`, and `Point`. These have a `_from_dict` class method, but this can just as well be a module-level function so the named tuple doesn't have to be subclassed.

On the other hand, I'm thinking of turning `MetricDescriptor` and `ResourceDescriptor` into ordinary classes. The former will eventually grow support for creation of custom metric descriptions.

I think the attributes of `MetricDescriptor` and `ResourceDescriptor` can perfectly well be writeable. This might in fact be useful in the case of `MetricDescriptor`.